workflow/controller/operator.go

```diff
@@ -495,6 +495,27 @@ func (woc *wfOperationCtx) countActivePods(boundaryIDs ...string) int64 {
 	return activePods
 }
 
+// countActiveChildren counts the number of active (Pending/Running) children nodes of parent parentName
+func (woc *wfOperationCtx) countActiveChildren(parentName string) int64 {
```

Comment:
I think countActiveChildren() should replace the use of countActivePods() and sum the number of all NodeTypePod, NodeTypeSteps, NodeTypeDAG nodes within a boundaryID as an aggregate. It can have similar logic to countActivePods() but modified slightly. Something like:

```
	activeChildren := 0
	for _, node := range woc.wf.Status.Nodes {
		if boundaryID != "" && node.BoundaryID != boundaryID {
			continue
		}
		switch node.Type {
		case wfv1.NodeTypePod, wfv1.NodeTypeSteps, wfv1.NodeTypeDAG:
		default:
			continue
		}
		switch node.Phase {
		case wfv1.NodePending, wfv1.NodeRunning:
			activeChildren++
		}
	}
	return activeChildren
```

Then, the existing calls to `countActivePods()` in `checkParallelism()` would be replaced with the call to `countActiveChildren()`.

We would then only use countActivePods() when checking against the global parallelism limit.

Reply:
Thanks! Finding children nodes with the same boundaryID should be able to solve my problem in https://github.com/argoproj/argo/pull/1046#discussion_r228700919. (I was stuck at using the `Children` field to find children 😅). I'll let you know if it works.

Reply:
It's working! 😀